pkg/kubelet/envvars/envvars.go

```diff
@@ -22,28 +22,30 @@ import (
 	"strconv"
 	"strings"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
 )
 
 // FromServices builds environment variables that a container is started with,
 // which tell the container where to find the services it may need, which are
-// provided as an argument.
-func FromServices(services []*v1.Service) []v1.EnvVar {
+// provided as an argument. "masterFilterFunc" checks if the service is the master
+// service and it can be of type ExternalName.
+func FromServices(services []*v1.Service, masterFilterFunc func(*v1.Service) bool) []v1.EnvVar {
```

Comment:
Could you share a little more about the advantages of specifying `masterFilterFunc` as an argument, instead of hard-coding it in the `FromServices` function? Could there ever be a time when the `masterFilterFunc` is different than the one you pass in `kubelet_pods.go`?

Reply:
The main reason is ease of testing. I can hard code it, but passing it feels more natural.
The actual implementation is tested in `kubelet_pods_test.go`

Reply:
I gotcha - agree the ease of testing is nice.

At the end of the day, don't feel super strongly.